unified-runtime/source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -140,20 +142,20 @@ ur_result_t ur_queue_immediate_in_order_t::queueGetNativeHandle(
     ur_queue_native_desc_t *pDesc, ur_native_handle_t *phNativeQueue) {
   std::ignore = pDesc;
   *phNativeQueue = reinterpret_cast<ur_native_handle_t>(
-      this->commandListManager.getZeCommandList());
+      this->commandListManager.get_no_lock()->getZeCommandList());
   return UR_RESULT_SUCCESS;
 }
 
 ur_result_t ur_queue_immediate_in_order_t::queueFinish() {
   TRACK_SCOPE_LATENCY("ur_queue_immediate_in_order_t::queueFinish");
 
   std::unique_lock<ur_shared_mutex> lock(this->Mutex);
```

Comment:
why do we still use this->mutex (here and in other places)? shouldn't we remove it and use command list locking exclusively?